tests/parser/mod.rs

```diff
@@ -200,22 +220,41 @@ fn match_expr_impl(e: &Expr, v: &Value) -> Result<()> {
         }
 
         #[cfg(feature = "rego-extensions")]
-        Expr::OrExpr { span, lhs, rhs } => {
+        Expr::OrExpr { span, lhs, rhs, .. } => {
             match_span_opt(span, &v["orexpr"]["span"])?;
             match_expr(lhs, &v["orexpr"]["lhs"])?;
             match_expr(rhs, &v["orexpr"]["rhs"])
         }
+    }?;
+
+    match (e.eidx(), &v["eidx"]) {
+        (eidx, Value::Number(n)) if n.as_u64() == Some(eidx as u64) => Ok(()),
+        _ => {
+            bail!(
+                "{}",
+                e.span().message(
+                    "mismatch-error",
+                    format!(
+                        "eidx mismatch: expected {:?}, got {:?}",
+                        v["eidx"],
+                        e.eidx(),
+                    )
+                    .as_str()
+                )
+            );
+        }
     }
 }
 
 fn match_expr(expr: &Expr, v: &Value) -> Result<()> {
-    match match_expr_impl(expr, v) {
+    /*match match_expr_impl(expr, v) {
         Ok(()) => Ok(()),
         Err(e) => bail!(
             "{e}\nexpr = {expr:#?}\nv={}\n-----------------------\n",
             serde_json::to_string_pretty(v)?
         ),
-    }
+    }*/
```

Comment:
same here, should this be uncommented or deleted?

Reply:
Good point. The commented-out code allows printing diff at each nesting level. I have changed it so that uncommenting the code will bring it back.
It's mainly for debugging test failures.